homeassistant/components/weather/__init__.py

```diff
@@ -346,6 +352,68 @@ def __init_subclass__(cls, **kwargs: Any) -> None:
                     cls.__name__,
                     report_issue,
                 )
+        if any(
+            method in cls.__dict__ for method in ("_attr_forecast", "forecast")
+        ) and not any(
+            method in cls.__dict__
+            for method in (
+                "async_forecast_daily",
+                "async_forecast_hourly",
+                "async_forecast_twice_daily",
+            )
+        ):
+            cls.__weather_legacy_forecast = True
+
+    @callback
+    def add_to_platform_start(
+        self,
+        hass: HomeAssistant,
+        platform: EntityPlatform,
+        parallel_updates: asyncio.Semaphore | None,
+    ) -> None:
+        """Start adding an entity to a platform."""
+        super().add_to_platform_start(hass, platform, parallel_updates)
+        _reported_forecast = False
+        if self.__weather_legacy_forecast and not _reported_forecast:
+            module = inspect.getmodule(self)
+            if module and module.__file__ and "custom_components" in module.__file__:
+                report_issue = "report it to the custom integration author."
+            else:
+                report_issue = (
+                    "create a bug report at "
+                    "https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue"
+                )
+            _LOGGER.warning(
+                (
+                    "%s::%s is using a forecast attribute on an instance of "
+                    "WeatherEntity, this is deprecated and will be unsupported "
+                    "from Home Assistant 2024.3. Please %s"
+                ),
+                self.__module__,
+                self.entity_id,
+                report_issue,
+            )
+            if report_issue.startswith("create"):
+                report_issue = (
+                    "create a [bug report]"
+                    "(https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue)"
+                )
+            ir.async_create_issue(
+                self.hass,
+                DOMAIN,
+                f"deprecated_weather_forecast_{self.platform.platform_name}",
+                breaks_in_ha_version="2024.3.0",
+                is_fixable=False,
+                is_persistent=False,
+                issue_domain=self.platform.platform_name,
+                severity=ir.IssueSeverity.WARNING,
+                translation_key="deprecated_weather_forecast",
+                translation_placeholders={
+                    "platform": self.platform.platform_name,
+                    "report_issue": report_issue,
```

Comment:
Have you tested that this works and is rendered as a link?
If it doesn't, there's a learn_more_url parameter too.

Reply:
Works like a charm
![image](https://github.com/home-assistant/core/assets/62932417/e1a1c5d5-09cd-4f5e-8ff9-7e4745d32f9a)